java-util/src/main/java/io/druid/java/util/http/client/pool/ChannelResourceFactory.java

```diff
@@ -109,6 +112,28 @@ public ChannelFuture generate(final String hostname)
 
       final ChannelPipeline pipeline = connectFuture.getChannel().getPipeline();
       pipeline.addFirst("ssl", sslHandler);
+      pipeline.addFirst("handler", new SimpleChannelUpstreamHandler()
```

Comment:
Thank you for catching this!

Reply:
I think this handler should be responsible for only connection errors. Every other errors should be handled by the handler in `NettyHttpClient` to properly notify callers on failures.

So, please rename the handler to be more intuitive like `connectionErrorHandler`. Also, this handler should be after `sslHandler` to catch all exceptions properly.

Reply:
Done.